smartapps/ady624/webcore-piston.src/webcore-piston.groovy

```diff
@@ -415,382 +167,383 @@ def pageClearAll() {
+	return [:]
 }
 
 Map pausePiston() {
 	state.active = false
-    def rtData = getRunTimeData()
+	def rtData = getRunTimeData()
 	def msg = timer "Piston successfully stopped", null, -1
-    if (rtData.logging) info "Stopping piston...", rtData, 0
-    checkVersion(rtData)
-    state.schedules = []
-    rtData.stats.nextSchedule = 0
-    unsubscribe()
-    unschedule()
-    app.updateSetting('dev', (Map) null)
-    app.updateSetting('contacts', (Map) null)
-    state.hash = null
-    state.trace = [:]
-    state.subscriptions = [:]
-    state.schedules = []
-    if (rtData.logging) info msg, rtData
-    updateLogs(rtData)
-    atomicState.schedules = []
+	if (rtData.logging) info "Stopping piston...", rtData, 0
+	checkVersion(rtData)
+	state.schedules = []
+	rtData.stats.nextSchedule = 0
+	unsubscribe()
+	unschedule()
+	app.updateSetting('dev', (Map) null)
+	app.updateSetting('contacts', (Map) null)
+	state.hash = null
+	state.trace = [:]
+	state.subscriptions = [:]
+	state.schedules = []
+	if (rtData.logging) info msg, rtData
+	updateLogs(rtData)
+	atomicState.schedules = []
 	atomicState.active = false
-    return rtData
+	return rtData
 }
 
 Map resume() {
 	state.active = true;
 	def tempRtData = getTemporaryRunTimeData()
-    def msg = timer "Piston successfully started", null,  -1
+	def msg = timer "Piston successfully started", null,  -1
 	if (tempRtData.logging) info "Starting piston... (${version()})", tempRtData, 0
-    def rtData = getRunTimeData(tempRtData, null, true)
-    checkVersion(rtData)
-    state.hash = null
-    state.subscriptions = [:]
-    atomicState.schedules = []
-    state.schedules = []
-    subscribeAll(rtData)
-    if (rtData.logging) info msg, rtData
-    updateLogs(rtData)
-    rtData.result = [active: true, subscriptions: state.subscriptions]
+	def rtData = getRunTimeData(tempRtData, null, true)
+	checkVersion(rtData)
+	state.hash = null
+	state.subscriptions = [:]
+	atomicState.schedules = []
+	state.schedules = []
+	subscribeAll(rtData)
+	if (rtData.logging) info msg, rtData
+	updateLogs(rtData)
+	rtData.result = [active: true, subscriptions: state.subscriptions]
 	atomicState.active = true
-    return rtData
+	return rtData
 }
 
 
 def setLoggingLevel(level) {
-    def logging = "$level".toString()
-    logging = logging.isInteger() ? logging.toInteger() : 0
-    if (logging < 0) logging = 0
-    if (logging > 3) logging = 3
+	def logging = "$level".toString()
+	logging = logging.isInteger() ? logging.toInteger() : 0
+	if (logging < 0) logging = 0
+	if (logging > 3) logging = 3
 	atomicState.logging = logging
-    return [logging: logging]
+	return [logging: logging]
 }
 
 def setCategory(category) {
 	atomicState.category = category
-    return [category: category]
+	return [category: category]
 }
 
 def test() {
 	handleEvents([date: new Date(), device: location, name: 'test', value: now()])
-    return [:]
+	return [:]
 }
 
 def execute(data, source) {
+	pause(250)
```

Comment:
Why the delay here? I have a piston that fires off other pistons acting as routines. With this it would add a quarter second delay before processing it, which is bad for time sensitive ones like motion, or lighting.

Reply:
this is a potential infinite loop if a set of pistons keep calling each other.  Again another method to slow down infinite loops due to the lack of overall execution timeout in HE.